lib/surface/compiler.ex

```diff
@@ -241,17 +241,20 @@ defmodule Surface.Compiler do
   defp convert_node_to_ast(:slot, {_, attributes, children, node_meta}, compile_meta) do
     meta = Helpers.to_meta(node_meta, compile_meta)
 
+    defined_slot_names =
+      meta.caller.module
+      |> Surface.API.get_slots()
+      |> Enum.map(& &1.name)
+
     # TODO: Validate attributes with custom messages
     name = attribute_value(attributes, "name", :default)
 
     index =
       attribute_value_as_ast(attributes, "index", %Surface.AST.Literal{value: 0}, compile_meta)
 
-    with true <- not is_nil(name) and is_atom(name),
```

Comment:
I removed this condition because the value of the name will always be fixed and will always be an atom with at least the a default value `:default` regarding the following function.

https://github.com/msaraiva/surface/blob/f373b0fbd193e9e60c95cbc42354e56346b031b6/lib/surface/compiler.ex#L425-L431

As the next line will also always return `:ok` I simplified the code. 

Feel free to tell me if I should revert this part.